### PR TITLE
Port the updateinfo generation code to createrepo_c

### PR DESCRIPTION
**Pull request by lmacken**
:dancers:

### CODE REVIEW

File: bodhi/metadata.py [hunk 1]
```diff
@@ -12,23 +12,23 @@
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-__version__ = '1.4'
+__version__ = '2.0'
```
Comment (ralphbean) [inline bodhi/metadata.py]: :moneybag:

File: bodhi/metadata.py [hunk 2]
```diff
@@ -12,23 +12,23 @@
...
 
 import os
-import glob
 import logging
 import shutil
 import tempfile
+import subprocess
```
Comment (ralphbean) [inline bodhi/metadata.py]: :sob:

File: bodhi/metadata.py [hunk 3]
```diff
@@ -170,181 +180,104 @@ def _fetch_updates(self):
...
             for rpm in rpms:
-                filename = "%s.%s.rpm" % (rpm['nvr'], rpm['arch'])
+                pkg = cr.UpdateCollectionPackage()
+                pkg.name = rpm['name']
+                pkg.version = rpm['version']
+                pkg.release = rpm['release']
+                pkg.epoch = rpm['epoch'] or '0'
+                pkg.arch = rpm['arch']
+
+                # TODO: how do we handle UpdateSuggestion.logout, etc?
```
Comment (ralphbean) [inline bodhi/metadata.py]: Maybe submit an RFE to createrepo_c?

Comment (ralphbean) [inline bodhi/metadata.py]: ... and link to the ticket from an inline comment here so future bodhi hackers will know?

File: bodhi/metadata.py (hunk 2 above)
Comment (lmacken) [inline bodhi/metadata.py]: Yeah, we ideally should be using the modifyrepo_c python interface

File: development.ini [hunk 4]
```diff
@@ -327,6 +327,11 @@ admin_packager_groups = provenpackager releng security_respons
 mandatory_packager_groups = packager
 
 ##
+## updateinfo.xml configuraiton
+##
+updateinfo_rights =
```
Comment (ralphbean) [inline development.ini]: Perhaps provide an example about what might go here?

File: bodhi/metadata.py (hunk 2 above)
Comment (ralphbean) [inline bodhi/metadata.py]: The actual usage of ``subprocess.check_call`` in the code here is not that bad.  It just references files that get written out to (not lots of arguments getting interpolated or anything like that).

File: bodhi/metadata.py (hunk 3 above)
Comment (lmacken) [inline bodhi/metadata.py]: Right now the metadata for this looks like `<reboot_suggested>True</reboot_suggested>`.

Sounds like we need a more flexible schema, maybe something like `<suggest type="logout"/>`

**Comment by ralphbean** [comment]
:+1:

**Comment by lmacken** [comment]
Thanks!

**Pull request merged**